buildJSON.js

```diff
@@ -0,0 +1,177 @@
+				const updatedFactQuery = convertFqlToGraphql(factQuery);
+
+				updatedFactMap.set(factName, updatedFactQuery);
+				// has to delete entry after completing transform operation
+				factMap.delete(factName);
+
+				conditionMap?.forEach((conditionValue, conditionKey) => {
+					if (conditionValue.includes(factName)) {
+						const conditionName = conditionKey;
+						let conditionQuery = conditionValue.replace(functionCallToReplace, '');
+
+						conditionQuery = conditionQuery.replace(`${factName}${functionVariables}`, `${factQuery}`)
+						conditionQuery = conditionQuery.replace(' ', '');
+						const updatedConditionQuery = convertFqlToGraphql(conditionQuery);
+
+						updatedConditionMap.set(conditionName, updatedConditionQuery);
+						// has to delete entry after completing transform operation
+						conditionMap.delete(conditionName);
+					}
+				});
+			}
+		});
+
+		// has to delete entry after completing transform operation
+		// so no duplicates will be present in the transformed map
+		objectMap.delete(objectName);
+	});
+
+	// After all parts of the JSON are created
+	// Merge them to return the full JSON
+	ruleMap?.forEach((ruleValue, ruleKey) => {
+
+		const ruleName = deCapitalizeFirstLetter(ruleKey.replace('Rule', ''));
+		let splittedString = ruleValue.split(' (');
+
+		// remove vars from rule string
+		splittedString.shift();
+
+		const bracketsAmount = splittedString.length;
+		let lastValue = splittedString[splittedString.length - 1];
+		lastValue = lastValue.substring(0, lastValue.length - bracketsAmount)
+		splittedString[splittedString.length - 1] = lastValue;
+
+		let tempLastKey = '';
+		let tempLastIndex = 0;
+
+		for (let i = 0; i < splittedString.length; i++){
+			const ruleEntry = splittedString[i];
+			const conditionSearchParam = ruleEntry.substring(0, ruleEntry.indexOf('('))
+
+			let entry = {};
+
+			// At top level the json will be created
+			// And filled with name and type
+			// only one all/any can be placed
+			if (i === 0) {
+				if (ruleEntry.includes('&&')) {
+					resultJSON = {
+						name: ruleName,
+						type: 'Rule',
+						all: [
+							updatedConditionMap.get(conditionSearchParam),
+						]
+					};
+					tempLastKey = 'all';
+				} else {
+					resultJSON = {
+						name: ruleName,
+						type: 'Rule',
+						any: [
+							updatedConditionMap.get(conditionSearchParam),
+						]
+					};
+					tempLastKey = 'any';
+				}
+			} else {
+				if (ruleEntry.includes('&&')) {
+					entry = createRuleEntry(ruleEntry, ' && ', updatedConditionMap);
+				} else {
+					entry = createRuleEntry(ruleEntry, ' || ', updatedConditionMap);
+				}
+
+				// Get current position in object
+				// Add the created entries
+				// Updated the index - if not first or last
+				if (i === splittedString.length - 1) {
+					tempLastKey.split('.').reduce((o,i)=> o[i], resultJSON).push(entry);
+				} else {
+					tempLastKey.split('.').reduce((o, i) => o[i], resultJSON).push(entry);
```

Comment:
line 128 and 130 has same code, it means it will run whether condition satisfy or not. SO move this line outside the condition